[CHERI-RISC-V] Turn trapping semantics on again

arichardson opened the pull request:
The latest CheriBSD release is not ready for these changes and has test failures due to these semantic changes. This will be turned on again in the future but we have to disable it for now as it prevents merges from QEMU dev to qemu-cheri.

arichardson committed:
[CHERI-RISC-V] Turn trapping semantics on again

The latest CheriBSD release is not ready for these changes and has test
failures due to these semantic changes. This will be turned on again in
the future but we have to disable it for now as it prevents merges from
QEMU dev to qemu-cheri.

jrtc27 commented on target/riscv/csr.c:
```diff
@@ -1342,7 +1342,7 @@ static int read_ccsr(CPURISCVState *env, int csrno, target_ulong *val)
     ccsr = set_field(ccsr, XCCSR_ENABLE, cpu->cfg.ext_cheri);
     ccsr = set_field(ccsr, XCCSR_DIRTY, 1); /* Always report dirty */
     /* Read-only feature bits. */
-    ccsr = set_field(ccsr, XCCSR_TAG_CLEARING, 1);
```
Might be better to split this out so the other half can be git revert'ed later?

arichardson committed:
[CHERI-RISC-V] Set XCCSR_TAG_CLEARING based on CHERI_TAG_CLEAR_ON_INVALID

We already have a pre-processor macro for the value of this bit and using
it directly avoids the two getting out of sync.

arichardson commented:
Test failure is a timeout - merging
